### DISCUSSION TITLE
Should Email Instructor cc the sender?

### DISCUSSION THREAD

**Discussion by Alex-Jordan** [openwebwork/webwork2#1542] [Ideas]
Sometimes a student would like to use Email Instructor to ask for help, and they have some work on paper they would like to share, but are unable or unwilling to type it up in the Email Instructor message.

Sometimes this leads to "help me; I do have some work but I can't share it with you in this message". Sometimes this leads to the student sending an email out side of WeBWorK with a photo of their work, and then the email doesn't have a link for the instructor to follow to speedily get back to the question.

I was thinking about how to make this situation better. Of course we could find a way to let them include attachments in their Email Instructor message, but maybe that is a big project. So I had the idea that at least if they were cc'd in their own message, then they could initiate the email thread with Email Instructor, but then reply to the message with a follow-up that has an attachment, using their regular email client.

Would this raise any red flags? (Do Email Instructor messages send information we don't actually want students to see?) Is there anything to gain by making "cc myself on this message" an option for the sender instead of just always happening by default?

**Comment by drgrice1** [comment]
I think it would be better to implement the capability of adding an attachment. That shouldn't be that hard. I could look into it. 

I don't think that cc'ing the student is the ideal solution.

**Reply by TimAlderson** [reply]
Attachments would be a great feature to have! 

Some immediate thoughts:

I don't see why one would opt out of allowing attachments, but perhaps there is an appetite for allowing opt-out?

It may also be helpful if those with professor level permissions could provide a custom text string of instructions in Course Configuration. This could be displayed say, as a byline to the attachment button or otherwise on the email page.  Such would allow instructors to provide course specific requirements regarding file types, sizes etc.   I have no idea how complicated it would be to add such an option, nor if there would be wide interest. Just a thought.

Regardless of the above, I think allowing attachments would be a fantastic step.

**Comment by drgrice1** [comment]
I have implemented attachments to feedback emails.  See pull request #1558.

I don't see a real reason for an opt-out option, but if there is demand for that it could be implemented.

The current implementation restricts students to attaching images, pdf files, csv files, zip files, and txt files, and restricts to a max filesize of 10 MBs.   The file size limitation is a server configuration option, so the server administrator can change this.  The student can only select files of the listed type in the file browser, and if a student attempts to upload to large of a file it will be rejected.
